### ISSUE TITLE
Release strategy

### ISSUE THREAD

**Issue by davidtgoldblatt** [jemalloc/jemalloc#1049]
We've been improving CI recently, and we're getting closer to a world in which the tip of dev is as well tested as a point release would be. We don't really have a firm idea how we want to handle this.

My initial inclination is to occasionally pick a commit we regard as particularly stable (say, it's gone a while with no perf anomalies in some FB production workloads), set master to point there, and start recommending external users to build off master. Then, in x.y.z form, x can be mallctl changes or breakages, y can be significant new features, and z can indicate merges into FreeBSD.

The distro workflow seems to be point-release focused, with some minor patches on top.

Another option would be to keep to a more traditional point-release mechanism, and release more frequently. I don't mind volunteering to do the changelogs / picking commits, but I don't think I'd be well situated to coordinate with the FreeBSD tree.

+cc @paravoid and @marxin, who cared about this in #761, and @jasone, BDFL.

**Comment by marxin** [comment]
For me as a distribution maintainer of the package, I would welcome to have releases as you have. Apart from that it's useful to have security fixes and serious bug fixes to land to a release branch.

That's what happened in #761. Fix is quite obvious, but we can't apply the patch as-is on top of 5.0.1 release. That's situation where a proper backport will be appreciated.

Thanks.

**Comment by kev009** [comment]
It's neat to see FreeBSD integrated into the release strategy, and hopefully we reciprocate good testing and feedback via all the ports build and run since this is our base malloc.  We're actively trying to figure out the C++11 in base issues, getting a good out-of-tree gcc solid has been a long time coming, as has clang on ppc and mips.

**Comment by paravoid** [comment]
Any movement here? Any plans to do a 5.0.2? I'm particularly interested in portability fixes, like e.g. #761, #979, #985, #999, but fixing any other known serious issues in 5.0.1 would be neat.

**Comment by davidtgoldblatt** [comment]
Yeah, I'll see what I can do for the portability stuff.

Does the tip of dev work on all the platforms you care about? I didn't realize that there were all-green builds yet.

**Comment by paravoid** [comment]
I haven't really tried that and there is no easy way for me to do so, as I'd have to login to all of the different kind of servers and manual build. The "right" way to do that would be to upload a new version to trigger a build on the autobuilder network; that can either be a new upstream version (e.g. 5.0.2-1), or a Debian revision (5.0.1-2) with custom patches applied on top of 5.0.1.

The only problem with the latter is that the fix for #761 doesn't apply cleanly on top of 5.0.1, so I'd have to port it myself and… it's just a bit suboptimal, esp. given that a) we generally dislike diverging from upstream b) Fedora packagers are also interested. Hopefully the last two points are getting us back on-topic for this issue :)

**Comment by davidtgoldblatt** [comment]
(BTW, I haven't forgotten about this, but there are a couple of performance pathologies that we discovered that we want to fix we cut another release).

**Comment by paravoid** [comment]
It's been a while, any news? :)

**Comment by davidtgoldblatt** [comment]
I think we'll be able to cut a release this quarter (probably the later part of it; we've got some long vacations coming in the early part). We've fixed the performance anomalies we were worried about (we think), and have only a few portability improvements that I want to get in before finalizing things.

**Comment by paravoid** [comment]
Sorry to be a nag -- are you still thinking of releasing a new version this quarter?

**Comment by davidtgoldblatt** [comment]
Yes -- order weeks is the plan. We might slip a little, but we shouldn't by a lot.

(Also, nagging is a a completely acceptable mechanism for helping us gauge issue priority as far as I'm concerned).

**Comment by kev009** [comment]
Are there plans in place to import to FreeBSD still or will @jasone do that?

**Comment by jasone** [comment]
@kev009, I will likely import 5.1 into FreeBSD.
